llvm/lib/Target/AArch64/MCTargetDesc/AArch64ELFStreamer.cpp

```diff
@@ -294,6 +379,58 @@ AArch64ELFStreamer &AArch64TargetELFStreamer::getStreamer() {
   return static_cast<AArch64ELFStreamer &>(Streamer);
 }
 
+void AArch64TargetELFStreamer::emitSubsection(unsigned Vendor,
+                                              unsigned IsMandatory,
+                                              unsigned ParameterType) {
+  StringRef VendorAsStr = ARMBuildAttrs::vendorToStr(Vendor);
+
+  // If exists, return.
+  for (MCELFStreamer::AttributeSubSection &SubSection : AttributeSubSections) {
+    if (SubSection.Vendor == VendorAsStr) {
+      llvm_unreachable("AArch64 build attributes subsection already exists");
```

Comment:
Is it not valid to switch back to a previous subsection, for example from user-written assembly directives?

Reply:
It is valid and possible to emit attribute to any subsection at any time, this achieved by the usage of a vector to hold the subsections.
There is no need to mechanically 'switch' between subsections. After a subsection has been created, it is always available and it is always possible to emit an attribute into it.

For example, user can first creates all the subsections that he wants, them add attribute to any of them at any time.

This function creates a new subsection and add it to the vector. 
The check here is for the case of creating the same subsection twice.

Reply:
Clarification: 
> It is valid and possible to emit attribute to any subsection at any time
Referring to the ELF mechanism. Asm attributes need to be emitted in order since they are being printed directly.

Reply:
This design does not require switching between sections in order to emit an ELF attribute to a specific section. As long as the required subsection has been created, an attribute can be emitted into it.

Reply:
It is valid. Programmatically, it is not required, attribute will be added to the correct subsection. When writing assembly it is done via adding the correct subsection header.